test-utils/src/lib.rs

```diff
@@ -0,0 +1,152 @@
+use portpicker::pick_unused_port;
+use rand::{distributions::Alphanumeric, Rng};
+use std::{
+    process::Command,
+    thread::sleep,
+    time::{Duration, SystemTime},
+};
+use uuid::Uuid;
+
+/// A docker instance for a postgres database. It should be used
+/// by tests as a singleton (e.g. once_cell::sync::Lazy), and any
+/// test logic that connects to it should separate cases by creating
+/// an unique database. Also, the instance should implement a destructor.
+///
+/// Example usage:
+///
+/// static PG: Lazy<PostgresDockerInstance> = Lazy::new(PostgresDockerInstance::default);
+
+/// #[dtor]
+/// fn cleanup() {
+///    PG.cleanup();
+/// }
+///
+///  #[tokio::test]
+/// async fn test_case() {
+///     // Create a unique database name so we have a new database for each test.
+///     let db_name = Uuid::new_v4().to_string();
+///     let db_uri = PG.get_unique_uri(db_name.as_str());
+///     
+///     // Test logic below, which can use `db_uri` to connect to the postgres instance.
+/// }
+///
+pub struct PostgresDockerInstance {
+    container_name: String,
+    base_uri: String,
+}
+
+impl Default for PostgresDockerInstance {
+    fn default() -> Self {
+        let s: String = rand::thread_rng()
+            .sample_iter(&Alphanumeric)
+            .take(7)
+            .map(char::from)
+            .collect();
+
+        PostgresDockerInstance::new(s.as_str())
+    }
+}
+
+impl PostgresDockerInstance {
```

Comment:
I just saw it today, looks promising.

Reply:
Opened https://github.com/shuttle-hq/shuttle/issues/1479. It shouldn't be difficult and might be a nice OSS contribution. What do you think?

Reply:
I haven't looked closely at test containers, but it does seem like it would simplify this a lot. For the GitHub issue we might want to expand a bit on what should change, and where, since we use this strategy in logger, provisioner and now auth. Provisioner does not use the code you moved to common-tests, but maybe we don't need to share code in common-tests if we use test containers.

Reply:
We might want to suggest that the contributor breaks it up into separate PRs, as well.

Reply:
Thanks @oddgrd ! I added more details under the definition of done for the associated issue. I missed that we have the docker tests in the `provisioner` too, and it will probably use `testcontainers` (not sure about the `common-tests` additional logic though). In `common-tests` we created API to create/get unique conn strings, referring to unique databases, that can be used in parallel by test cases. I think this is the outstanding part shared by `logger` and `auth` soon. Not sure if the `provisioner` needs this functionality for its integration tests, and if not, it can probably avoid using `common-tests`, integrating directly with `testcontainers`.